From 5451a4b8f67d4597f524f6d9e85a0bdeacd89a3f Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 9 Mar 2017 15:09:22 +0200
Subject: [PATCH 1452/2241] dts: a70x0: a80x0: Update the sdhci compatibility
 strings

Update SDHCI interface compatibility strings according to
Xenon driver v6 code.
Remove obsolete "slow mode" and "max speed" parameters from
AP SDHCI DTS - all AP SDHCI limitations are handled by Xenion
driver according to compatibility string.

Change-Id: I2c432515200eb8cc4842e3b08362c8d77fa6ca88
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37358
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi | 4 +---
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 2 +-
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 1f63ff2..7c10692 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -341,7 +341,7 @@
 			};
 			/* The SDIO should not be enabled in Z1 DSTI/DST */
 			sdhci@6e0000 {
-				compatible = "marvell,armada8k-sdhci";
+				compatible = "marvell,armada-ap806-sdhci";
 				reg = <0x6e0000 0x300>;
 				interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
 				clock-names = "core";
@@ -350,8 +350,6 @@
 				 * The AP SDIO interface should work in PHY slow mode
 				 * and clock not higher than 50MHz due to HW limitation
 				 */
-				marvell,xenon-phy-slow-mode;
-				max-frequency = <50000000>;
 				dma-coherent;
 				status = "disabled";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 62a82ea..bf7abe8 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -310,7 +310,7 @@ mdio@15b000 {
 };
 
 sdhci@780000 {
-	compatible = "marvell,armada8k-sdhci";
+	compatible = "marvell,armada-cp110-sdhci";
 	reg = <0x780000 0x300>;
 	interrupts = <ICU_GRP_NSR 27 IRQ_TYPE_LEVEL_HIGH>;
 	clock-names = "core";
-- 
2.7.4

